nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,297 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.epgstation;
+
+  username = config.users.users.epgstation.name;
+  groupname = config.users.users.epgstation.group;
+
+  settingsFmt = pkgs.formats.json {};
+  settingsTemplate = settingsFmt.generate "config.json" cfg.settings;
+  preStartScript = let
+    readPassword = f: if (f != null)
+      then "$(head -n1 \"${f}\")"
+      else "INSECURE_GO_CHECK_CONFIGURATION_NIX";
+  in
+  pkgs.writeScript "epgstation-prestart" ''
+    #!${pkgs.runtimeShell}
+
+    PASSWORD="${readPassword cfg.basicAuth.passwordFile}"
+    DB_PASSWORD="${readPassword cfg.database.passwordFile}"
+
+    # setup configuration
+    sed \
+      -e "s,@password@,$PASSWORD,g" \
+      -e "s,@dbPassword@,$DB_PASSWORD,g" \
+      ${settingsTemplate} > /etc/epgstation/config.json
+    chmod 440 /etc/epgstation/config.json
+    chown "${username}:${groupname}" /etc/epgstation/config.json
+
+    # XXX: Use password authentication, since mysqljs does not yet support auth_socket
+    if [ ! -e /var/lib/epgstation/db-created ]; then
+      ${pkgs.mysql}/bin/mysql -e \
+        "GRANT ALL ON ${cfg.database.name}.* TO ${username}@localhost IDENTIFIED by \"$DB_PASSWORD\";"
```

Comment:
Could you explain what this does? I'm not very familiar with mysql, so its hard for me to judge the security/correctness.

Reply:
It creates a database accessible by the given username and password. I remember seeing more of these in nixpkgs a while back, but it seem to have been replaced with unix domain socket authentication.

https://github.com/NixOS/nixpkgs/search?q=grant+all+on&type=Code